src/brod_client.erl

```diff
@@ -144,9 +144,9 @@ get_producer(Client, Topic, Partition) ->
   case get_partition_worker(Client, ?PRODUCER_KEY(Topic, Partition)) of
     {ok, Pid} ->
       {ok, Pid};
+    {error, client_down} = Error ->
+      maybe_start_producer(Client, Topic, Partition, Error);
```

Comment:
Do we need to restart the producer regardless of `auto_start_producers`

Reply:
if client is down, the maybe_start_producer call is unlikely to succeed ?

Reply:
if client is down, the maybe_start_producer call is unlikely to succeed ?